### PR TITLE
add a lanugage arg for compile

### LINKED ISSUE #93: [CT-1163] Add `language` to `RPCExecParameters` to support interactive compilation for dbt Python
The `compile_sql` method should accept a `language` argument, with two valid options: `("sql", "python")`. (We should do the same things for `run_sql`, for consistency.)

I think this would require a change to `RPCExecParameters`:

https://github.com/dbt-labs/dbt-rpc/blob/b45d09e487fedb1b12fd3214081e593d52332bb4/dbt_rpc/contracts/rpc.py#L50-L54

If the argument is provided, we should (somewhere) set the `ModelLanguage` attribute of the RPC node to be the language provided. That should be enough for it to be compiled properly [here in `dbt-core`](https://github.com/dbt-labs/dbt-core/blob/1db48b3cca25912977a753b54d1b69729ccb245b/core/dbt/compilation.py#L371-L389).

In order for this to not be a breaking change, when running with earlier versions of `dbt-core` (<1.3), we might need to check if the node `hasattr` for language already. If it does need to be a breaking change, we'll just bump `dbt-rpc` to minor version 0.2.0.

We shouldn't go through the trouble of renaming and aliasing these methods because `dbt-rpc` is soon to be deprecated.

### PR DESCRIPTION
**Pull request by ChenyuLInx**
Resolve #93

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-core&subdirectory=core
-git+https://github.com/dbt-labs/dbt-core.git#egg=dbt-postgres&subdirectory=plugins/postgres
+git+https://github.com/dbt-labs/dbt-core.git@rpc_support_python_compile#egg=dbt-core&subdirectory=core
```
Comment (ChenyuLInx) [inline dev-requirements.txt:1]: remove before merge

**Review by jtcohen6 — COMMENTED** [review]
I started a review a few days ago and failed to submit ... my bad!

File: dbt_rpc/parser/rpc.py [hunk 2]
```diff
@@ -41,8 +41,8 @@ def get_compiled_path(cls, block: FileBlock):
 
         return os.path.join('rpc', block.name)
 
-    def parse_remote(self, sql: str, name: str) -> ParsedRPCNode:
-        source_file = SourceFile.remote(sql, self.project.project_name)
+    def parse_remote(self, sql: str, name: str, language: Optional[str] = 'sql') -> ParsedRPCNode:
+        source_file = SourceFile.remote(sql, self.project.project_name, language)
```
[the remark this answers is not in the document]
Comment (leahwicz) [reply dbt_rpc/parser/rpc.py:45]: So no RPC code change at all? Or change it in a different way to allow back compat? 

If we optional/default to `sql` (like this is) could this not be back compat with dbt versions <1.2 and release dbt-rpc 1.1.3 instead?

[the remark this answers is not in the document]
Comment (ChenyuLInx) [reply dbt_rpc/parser/rpc.py:45]: I don't think there's any more changes in dbt-rpc need to happen to support previous versions of dbt-core. Given that all we did was add the file extension to the remote file object we created for it and reused logic core has(different parsing/compilation path given different file name).
With that being said, we also won't raise any error if you are doing this with core versions pre 1.3. The language argument would just be ignored and we compile things as sql(could be confusing, but the result should also be obvious enough).

What do we think? Maybe check the version and through an error if user tries to specify language in dbt-rpc running with dbt-core pre 1.3?

[the remark this answers is not in the document]
Comment (gshank) [reply dbt_rpc/parser/rpc.py:45]: I'm okay with ignoring it.

**Review by gshank — APPROVED** [review]

**Pull request merged**